### PR TITLE
Pass underlying NSError localizedDescription

### PR DESCRIPTION
**Pull request by taquitos**
If extra context is nil, use the underlying NSError's localizationDescription

### CODE REVIEW

**Review by joshdholtz — APPROVED** [review]
Seems good to me 💪

File: Purchases/FoundationExtensions/Error+Extensions.swift [hunk 1]
```diff
@@ -27,9 +27,10 @@ extension Error {
         }
 
         let asNSError = self as NSError
+        let underlyingNSError = attachedError as NSError
```
Comment (NachoSoto) [inline Purchases/FoundationExtensions/Error+Extensions.swift:30]: Maybe move this to the guard? `attachedError` isn't used:
```swift
guard let underlyingError = maybeError as NSError? else { }
```

File: Purchases/FoundationExtensions/Error+Extensions.swift [hunk 2]
```diff
@@ -27,9 +27,10 @@ extension Error {
...
         var userInfo = asNSError.userInfo as [NSError.UserInfoKey: Any]
-        userInfo[NSUnderlyingErrorKey as NSString] = attachedError
-        userInfo[ErrorDetails.extraContextKey] = extraContext
+        userInfo[NSUnderlyingErrorKey as NSString] = underlyingNSError
+        userInfo[ErrorDetails.extraContextKey] = extraContext ?? underlyingNSError.localizedDescription
```
Comment (NachoSoto) [inline Purchases/FoundationExtensions/Error+Extensions.swift:33]: I like it.

**Review by NachoSoto — APPROVED** [review]

File: Purchases/FoundationExtensions/Error+Extensions.swift (hunk 1 above)
Comment (taquitos) [reply Purchases/FoundationExtensions/Error+Extensions.swift:30]: 🤦‍♂️. Thank you

File: Purchases/FoundationExtensions/Error+Extensions.swift (hunk 2 above)
Comment (taquitos) [reply Purchases/FoundationExtensions/Error+Extensions.swift:32]: 🕺

**Pull request merged**